### DISCUSSION TITLE
Easier way to update translation file

### DISCUSSION THREAD

**Discussion by JDFS404** [jflamy/owlcms4#824] [Q&A] [answered]
I want to have an easier and non-cumbersome way of updating the translation. I notice that whenever Docker updates my version, and some new fields are included, the translation file needs to be updated as well to prevent this:
![CleanShot 2024-01-12 at 11 14 05@2x](https://github.com/jflamy/owlcms4/assets/26678110/0f4a9f43-5cfd-47bf-9e97-468f3cf2e5da)
Or perhaps I'm doing something wrong? But what I usually do, is get the newest translation file from your `local.zip` on GitHub and include that in my own `local.zip` file and re-upload the whole shebang.

**Comment by JDFS404** [comment]
I'm using the English version btw.

**Comment by jflamy** [comment]
The packaged version running in the cloud has the correct translation file always, so you don't actually need to update the translation file ever.

If you don't change the translations, don't include the i18n directory in your local.zip, just include the directories you do change.

The best way to proceed if you are doing customizations is as follows:
- update your local version to match what you will be running in the cloud.
- check that all is good -- currently there is a fair amount of movement with the css files, and if you changed them you definitely need to check locally.
- if all is fine on your local setup, then zipping all your local directory should work because the versions match.
- **But** it is always safer to only include the directories you changed.  So you can create the full local zip, open it and delete what you haven't touched.

**Comment by jflamy** [comment]
The idea for local.zip is that a federation would create their own set of templates and css and flags and whatever, then the member clubs would upload that and it would override what is installed.
In the cloud, if the server is restarted, the only thing that sticks around is the database.  Saving the dozens of individual files would be impractical, so I save the local.zip file and unpack it when the server starts.  The files in the local directory so created are given priority over what is in the actual distribution, so only additional or changed items need to be in the zip.

**Comment by JDFS404** [comment]
Interesting! This means I can get rid of all the template files which I never touch? I only change the Records files - would that mean my local file would be very small with only 1 folder?

**Comment by jflamy** [comment]
You don't even need to send the records, because the records management page uploads the records to the database -- this has been the case for quite a while.  So you don't actually need a local zip.
https://owlcms.github.io/owlcms4/#/2500RecordsManagement?id=loading-official-records

**Reply by JDFS404** [reply]
That's correct - however we have some comps which either don't use OWLCMS or use their own OWLCMS instance. So I update those records myself in my 'own' database. Or is there an easier way?

**Comment by jflamy** [comment]
Local is used for things that I can't easily fit in the database (css, templates, etc.)   There is a request open to load the pictures from an upload, which would allow taking the picture of the athlete at the weigh-in.

**Reply by jflamy** [reply]
For a second I thought you were USA and then remembered you were NL.

**Comment by jflamy** [comment, marked as answer]
Keeping an official master Excel up-to-date in owlcms format is the easiest way.   
Competitions that use owlcms load that file before the competition -- we keep a copy in a Google Drive.  At the end of a competition, there is a report on the records management page that gives you all the new records, so they can send you that -- if you just append to the master Excel it will work, then you can sort as you want.
Competitions that don't use owlcms you just update the master Excel manually.

**Reply by JDFS404** [reply]
That sounds exactly like how I do it now! My route is as follows:

1. Extract Final Results;
2. Paste into a Google Sheets Template I made, where all women and men are split per Q-points in 2024;
3. Data gets automatically pulled in via `IMPORTRANGE` in another spreadsheet which checks for new Records per class per age category;
4. That Sheet updates a Record file which has the same format as OWLCMS;
5. That Record file can be downloaded by anybody within my federation to use in OWLCMS.

Nice to hear how others are tackling this 'issue'!
